### PR TITLE
fix: add proper space around linear meter items

### PR DESCRIPTION
**Pull request by dlockhart**
Linear meters are relying on a trailing `&nbsp;` to insert space between the primary and secondary text items. This doesn't work in RTL:

Screenshot 2024-08-01 at 3 49 46 PM

It also results in an undesired extra space next to the text in LTR when there's only primary text (highlighted in red):

Screenshot 2024-08-01 at 3 51 23 PM

To fix this, I'm going to use `flexbox`'s `gap`. I'll first put up some new vdiffs to highlight the problem -- here's the report from that:
https://vdiff.d2l.dev/BrightspaceUI/core/716fb738df7570376328eda38c6e12630ea9dbbb/2024-08-01_19-59-42/report/

And then here's the report with the fix. Note that because `&nbsp;` became `0.45rem` most of the other diffs were affected in a minor way.
https://vdiff.d2l.dev/BrightspaceUI/core/69dc60cf5b8f7d835c4327942cbe9f12f0bf1e7c/2024-08-01_20-14-48/report/

### CODE REVIEW

File: components/meter/meter-linear.js [hunk 1]
```diff
@@ -25,25 +25,16 @@ class MeterLinear extends MeterMixin(RtlMixin(LitElement)) {
 			}
 
 			:host > div {
-				display: block;
+				display: flex;
+				flex-direction: column;
+				gap: 0.45rem;
```
Comment (dlockhart) [inline components/meter/meter-linear.js:30]: Was able to apply `flexbox` to both the stacked and inline cases, and use the same gap for both which eliminated the need for the bottom margin further on.

File: components/meter/meter-linear.js [hunk 2]
```diff
@@ -25,25 +25,16 @@ class MeterLinear extends MeterMixin(RtlMixin(LitElement)) {
...
 			}
 
 			:host([text-inline]) > div {
 				align-items: center;
-				display: flex;
 				flex-direction: row;
 			}
 
-			:host([text-inline]) .d2l-meter-linear-full-bar {
```
Comment (dlockhart) [inline components/meter/meter-linear.js:37]: Not needed now that we're using flexbox for the stacked case.

File: components/meter/meter-linear.js [hunk 3]
```diff
@@ -64,14 +54,11 @@ class MeterLinear extends MeterMixin(RtlMixin(LitElement)) {
 
 			.d2l-meter-linear-inner-bar {
 				background-color: var(--d2l-color-celestine);
-				left: 0;
+				inset-inline-start: 0;
```
Comment (dlockhart) [inline components/meter/meter-linear.js:57]: Spring cleaning.

**Review by bearfriend — APPROVED** [review]
Initially I was concerned about text like "Visted" and "40%" being spaced apart with css instead of an actual text space, but now that I look at it this is better because the primary and secondary text should be separate and are not meant to be read together as a sentence. The increased space between helps make that evident too, and it's potentially a localization win 🎉

**Pull request merged**